### PR TITLE
core: `filter_code`

### PR DESCRIPTION
**Pull request by spolu**
## Description

Adds a `filter_code` field to the specification of data source blocks. If specified and if returns a non empty value, parse it as a `SearchFilter` and then attempt to merge the one we got from the config of the app call.

Merge merges all arrays and overrides `timestamp.gt\lt`. If no code is specified or returns null/undefined, the behavior of merging on top of this empty SearchFilter the one we get from configurations preserves the behavior. 

Tests:
- [x] take an old app and check that the move does not break the data source block
- [x] add a filter_code on that old app and check that everything works as expected
- [x] build a fresh data_source block and check that everything works as expected.

Screenshot from 2024-03-26 14-01-50

## Risk

Messing with existing apps. Will check our assistant with RAG are not impacted post deploy.

## Deploy Plan

- deploy `core`
- test existing assistants
- deploy `front`
- test existing assistants

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
Looks great

File: core/src/blocks/data_source.rs [hunk 1]
```diff
@@ -233,12 +238,47 @@ impl Block for DataSource {
...
+                    {
+                        Ok((v, l)) => (v, l),
+                        Err(e) => Err(anyhow!("Error in filter code: {}", e))?,
+                    };
+
+                (
+                    match filter_value {
+                        Value::Null => None,
+                        Value::Object(o) => {
+                            println!("FILTER CODE OUTPUT: {:?}", o);
```
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/blocks/data_source.rs:263]: will remove :+1:

File: front/lib/config.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import type { BlockRunConfig, SpecificationType } from "@dust-tt/types";
 
 export function extractConfig(spec: SpecificationType): BlockRunConfig {
+  console.log("extractConfig", spec);
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/config.ts:4]: will remove :+1:

File: core/src/blocks/data_source.rs [hunk 3]
```diff
@@ -233,12 +238,47 @@ impl Block for DataSource {
...
+                let (filter_value, filter_logs): (Value, Vec<Value>) =
+                    match tokio::task::spawn_blocking(move || {
+                        let mut script = Script::from_string(filter_code.as_str())?
+                            .with_timeout(std::time::Duration::from_secs(10));
+                        script.call("_fun", &e)
+                    })
+                    .await?
+                    {
+                        Ok((v, l)) => (v, l),
+                        Err(e) => Err(anyhow!("Error in filter code: {}", e))?,
```
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/blocks/data_source.rs:256]: Will try to optimize :+1:

**Pull request merged**